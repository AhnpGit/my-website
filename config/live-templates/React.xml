<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
