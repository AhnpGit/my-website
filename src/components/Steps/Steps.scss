@import '../../assets/sass/abstracts/functions';
@import '../../assets/sass/abstracts/variables';
@import '../../assets/sass/abstracts/mixins';

.wrapper__steps {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  @include mq(nth($medium-up, 1)) {
    padding: 0 map_get($base, unit);
  }

  .steps__intro {
    @extend .wrapper__steps;
    padding: 0;

    h2 {
      text-align: center;
      word-spacing: 2px;
      letter-spacing: 1px;

      &:after {
        display: block;
        height: 2px;
        background-color: map_get($colors, tahitiGold);
        content: " ";
        width: 100px;
        margin: 0 auto;
        margin-top: map_get($base, unit);
      }
    }
  }

  .steps__description {
    @extend .wrapper__steps;
    padding: 0;

    @include mq(nth($medium-up, 1)) {
      flex-direction: row;
    }

    .steps__image {
      width: 40%;
      margin-bottom: map_get($base, 'unit');

      @include mq(nth($medium-only, 1)) {
        width: 25%;
      }

      @include mq(nth($large-up, 1)) {
        width: 20%;
      }
    }
  }
}

.steps__items {
  @extend .wrapper__steps;
  padding: 0;

  align-items: flex-start;
  @include mq(nth($medium-up, 1)) {
    margin-left: map_get($base, 'unit');
  }

  .steps__item {
    display: flex;
    align-items: center;
    margin-bottom: map_get($base, 'unit');

    p {
      margin-bottom: 0;
    }

    div {
      color: map_get($colors, 'tahitiGold');
      font-size: 150%;
      height: 54px;
      width: 54px;
      border: 2px solid #e67e22;
      border-radius: 50%;
      display: inline-block;
      text-align: center;
      margin-right: map_get($base, 'unit');
      padding: 6px 11px;

      @include mq(nth($small-only, 1)) {
        height: 40px;
        width: 40px;
        margin-right: 10px;
        font-size: 120%;
      }
    }

    .steps__item__link {
      height: map-deep-get($base, 'icon', 'large');
      @include on-event {
        border-bottom: none;
      }
    }

    svg {
      width: 100%;
      height: 100%;
      fill: black;
    }
  }
}


