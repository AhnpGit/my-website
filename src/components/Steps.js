import React from 'react'

const Steps = () => (
    <section className="section-steps" id="works">
        <div className="row">
            <h2>How it works &mdash; Simple as 1, 2, 3</h2>
        </div>
        <div className="row">
            <div className="col span-1-of-2 steps-box">
                <img src={require(`../assets/img/app-iPhone.png`)} alt="Omnifood app on iPhone" className="app-screen" />
            </div>
            <div className="col span-1-of-2 steps-box">
                <div className="works-step clearfix">
                    <div>1</div>
                    <p>Choose the subscription plan that best fits your needs and sign up today.</p>
                </div>
                <div className="works-step clearfix">
                    <div>2</div>
                    <p>Order your delicious meal using our mobile app or website. Or you can even call us!</p>
                </div>
                <div className="works-step clearfix">
                    <div>3</div>
                    <p>Enjoy your meal after less than 20 minutes. See you the next time!</p>
                </div>

                <a href="" className="btn-app"><img src={require(`../assets/img/download-app.svg`)} alt="App Store Button" /></a>
                <a href="" className="btn-app"><img src={require(`../assets/img/download-app-android.png`)} alt="Play Store Button" /></a>
            </div>
        </div>
    </section>
)

export default Steps