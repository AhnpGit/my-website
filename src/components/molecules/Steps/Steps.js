// @flow

import React from 'react'
import Icon from '../../atoms/Icon/Icon'

/**
 * Steps component displays an image, some text blocks and 2 icons
 * @constructor
 */
const Steps = () => (
  <section className="steps">
    <div className="steps__intro">
      <h2>How it works &mdash; Simple as 1, 2, 3</h2>
    </div>
    <div className="steps__description">
      <div className="steps__image">
        <img src="https://lh6.googleusercontent.com/CZHrY4sPHrtka5iLVGyQYDMK3JSUYakzUDUv4EC6GEpYqeBZMaygcg" alt="Omnifood app on iPhone" />
      </div>
      <div className="steps__items">
        <div className="steps__item">
          <div>1</div>
          <p>Choose the subscription plan that best fits your needs and sign up today.</p>
        </div>
        <div className="steps__item">
          <div>2</div>
          <p>Order your delicious meal using our mobile app or website. Or you can even call us!</p>
        </div>
        <div className="steps__item">
          <div>3</div>
          <p>Enjoy your meal after less than 20 minutes. See you the next time!</p>
        </div>
        <div className="steps__item">
          <a href="" className="steps__item__link">
            <Icon name="app-store-badge" />
          </a>
          <a href="" className="steps__item__link">
            <Icon name="google-play-badge" />
          </a>
        </div>
      </div>
    </div>
  </section>
)

export default Steps
