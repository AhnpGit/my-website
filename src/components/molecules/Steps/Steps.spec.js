import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Steps from './Steps'

describe('Steps component', () => {
  it('renders correctly without any issues', () => {
    const component = shallow(
      <Steps />
    )
    const result = shallowToJson(component)

    expect(result).toMatchSnapshot()
  })
})
