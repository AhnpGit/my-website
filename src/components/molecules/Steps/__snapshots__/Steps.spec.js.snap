// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps component renders correctly without any issues 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="steps"
>
    <div
        className="steps__intro"
    >
        <h2>
            How it works — Simple as 1, 2, 3
        </h2>
    </div>
    <div
        className="steps__description"
    >
        <div
            className="steps__image"
        >
            <img
                alt="Omnifood app on iPhone"
                src="https://drive.google.com/uc?id=0BwwULdUQlf8mVkFjdkw4OVpfRVk"
            />
        </div>
        <div
            className="steps__items"
        >
            <div
                className="steps__item"
            >
                <div>
                    1
                </div>
                <p>
                    Choose the subscription plan that best fits your needs and sign up today.
                </p>
            </div>
            <div
                className="steps__item"
            >
                <div>
                    2
                </div>
                <p>
                    Order your delicious meal using our mobile app or website. Or you can even call us!
                </p>
            </div>
            <div
                className="steps__item"
            >
                <div>
                    3
                </div>
                <p>
                    Enjoy your meal after less than 20 minutes. See you the next time!
                </p>
            </div>
            <div
                className="steps__item"
            >
                <a
                    className="steps__item__link"
                    href=""
                >
                    <Icon
                        name="app-store-badge"
                    />
                </a>
                <a
                    className="steps__item__link"
                    href=""
                >
                    <Icon
                        name="google-play-badge"
                    />
                </a>
            </div>
        </div>
    </div>
</section>,
  "nodes": Array [
    <section
      className="steps"
>
      <div
            className="steps__intro"
      >
            <h2>
                  How it works — Simple as 1, 2, 3
            </h2>
      </div>
      <div
            className="steps__description"
      >
            <div
                  className="steps__image"
            >
                  <img
                        alt="Omnifood app on iPhone"
                        src="https://drive.google.com/uc?id=0BwwULdUQlf8mVkFjdkw4OVpfRVk"
                  />
            </div>
            <div
                  className="steps__items"
            >
                  <div
                        className="steps__item"
                  >
                        <div>
                              1
                        </div>
                        <p>
                              Choose the subscription plan that best fits your needs and sign up today.
                        </p>
                  </div>
                  <div
                        className="steps__item"
                  >
                        <div>
                              2
                        </div>
                        <p>
                              Order your delicious meal using our mobile app or website. Or you can even call us!
                        </p>
                  </div>
                  <div
                        className="steps__item"
                  >
                        <div>
                              3
                        </div>
                        <p>
                              Enjoy your meal after less than 20 minutes. See you the next time!
                        </p>
                  </div>
                  <div
                        className="steps__item"
                  >
                        <a
                              className="steps__item__link"
                              href=""
                        >
                              <Icon
                                    name="app-store-badge"
                              />
                        </a>
                        <a
                              className="steps__item__link"
                              href=""
                        >
                              <Icon
                                    name="google-play-badge"
                              />
                        </a>
                  </div>
            </div>
      </div>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Steps />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="steps"
>
          <div
                    className="steps__intro"
          >
                    <h2>
                              How it works — Simple as 1, 2, 3
                    </h2>
          </div>
          <div
                    className="steps__description"
          >
                    <div
                              className="steps__image"
                    >
                              <img
                                        alt="Omnifood app on iPhone"
                                        src="https://drive.google.com/uc?id=0BwwULdUQlf8mVkFjdkw4OVpfRVk"
                              />
                    </div>
                    <div
                              className="steps__items"
                    >
                              <div
                                        className="steps__item"
                              >
                                        <div>
                                                  1
                                        </div>
                                        <p>
                                                  Choose the subscription plan that best fits your needs and sign up today.
                                        </p>
                              </div>
                              <div
                                        className="steps__item"
                              >
                                        <div>
                                                  2
                                        </div>
                                        <p>
                                                  Order your delicious meal using our mobile app or website. Or you can even call us!
                                        </p>
                              </div>
                              <div
                                        className="steps__item"
                              >
                                        <div>
                                                  3
                                        </div>
                                        <p>
                                                  Enjoy your meal after less than 20 minutes. See you the next time!
                                        </p>
                              </div>
                              <div
                                        className="steps__item"
                              >
                                        <a
                                                  className="steps__item__link"
                                                  href=""
                                        >
                                                  <Icon
                                                            name="app-store-badge"
                                                  />
                                        </a>
                                        <a
                                                  className="steps__item__link"
                                                  href=""
                                        >
                                                  <Icon
                                                            name="google-play-badge"
                                                  />
                                        </a>
                              </div>
                    </div>
          </div>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="steps"
>
          <div
                    className="steps__intro"
          >
                    <h2>
                              How it works — Simple as 1, 2, 3
                    </h2>
          </div>
          <div
                    className="steps__description"
          >
                    <div
                              className="steps__image"
                    >
                              <img
                                        alt="Omnifood app on iPhone"
                                        src="https://drive.google.com/uc?id=0BwwULdUQlf8mVkFjdkw4OVpfRVk"
                              />
                    </div>
                    <div
                              className="steps__items"
                    >
                              <div
                                        className="steps__item"
                              >
                                        <div>
                                                  1
                                        </div>
                                        <p>
                                                  Choose the subscription plan that best fits your needs and sign up today.
                                        </p>
                              </div>
                              <div
                                        className="steps__item"
                              >
                                        <div>
                                                  2
                                        </div>
                                        <p>
                                                  Order your delicious meal using our mobile app or website. Or you can even call us!
                                        </p>
                              </div>
                              <div
                                        className="steps__item"
                              >
                                        <div>
                                                  3
                                        </div>
                                        <p>
                                                  Enjoy your meal after less than 20 minutes. See you the next time!
                                        </p>
                              </div>
                              <div
                                        className="steps__item"
                              >
                                        <a
                                                  className="steps__item__link"
                                                  href=""
                                        >
                                                  <Icon
                                                            name="app-store-badge"
                                                  />
                                        </a>
                                        <a
                                                  className="steps__item__link"
                                                  href=""
                                        >
                                                  <Icon
                                                            name="google-play-badge"
                                                  />
                                        </a>
                              </div>
                    </div>
          </div>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Steps />,
}
`;
